orderBook.js

```diff
@@ -0,0 +1,43 @@
+
+const reconcileOrder = (existingBook, incomingOrder) => {
+  let updatedBook = []
+  if (existingBook.length === 0 ||
+    existingBook[0].type === incomingOrder.type ||
+    existingBook[0].quantity != incomingOrder.quantity && existingBook[0].price != incomingOrder.price) {
```

Comment:
You can fulfill orders of different quantities so that check shouldn't be in here